crates/merod/src/cli/config.rs

```diff
@@ -44,53 +64,291 @@ impl FromStr for KeyValuePair {
+        } else {
+            eprintln!("warning: no changes were made to the configuration");
+        }
+
+        Ok(())
+    }
+
+    pub async fn validate_toml(&self, doc: &toml_edit::DocumentMut) -> EyreResult<()> {
+        let temp_file = tempfile::NamedTempFile::new()?;
+        let temp_path = temp_file.path().to_owned();
+
+        write(&temp_path, doc.to_string()).await?;
+
+        let temp_path_utf8 = Utf8PathBuf::try_from(temp_path)
+            .map_err(|_| eyre!("Failed to convert temp path to UTF-8"))?;
+
+        ConfigFile::load(&temp_path_utf8)
+            .await
+            .map_err(|e| eyre!("Config validation failed: {}", e))?;
+
+        temp_file.close()?;
+
+        Ok(())
+    }
+
+    fn print_hints(&self, hints: &[KeyValuePair]) -> EyreResult<()> {
+        let mut table = Table::new();
+        table.load_preset("││──├─┤─┼─└ ┴┬┌ ┐");
+        table.set_header(vec![
+            Cell::new("Key").fg(ComfyColor::Blue),
+            Cell::new("Type").fg(ComfyColor::Yellow),
+            Cell::new("Description").fg(ComfyColor::Green),
+        ]);
 
-            current[key_parts[key_parts.len() - 1]] = Item::Value(kv.value.clone());
+        for kv in hints {
+            let key = kv.key.trim_end_matches('?');
+            if let Some(schema) = CONFIG_SCHEMA.find(key) {
+                table.add_row(vec![
+                    Cell::new(key),
+                    Cell::new(schema.type_info),
+                    Cell::new(schema.description),
+                ]);
+
+                for child in &schema.children {
+                    let child_key = format!("{}.{}", key, child.path);
+                    table.add_row(vec![
+                        Cell::new(child_key),
+                        Cell::new(child.type_info),
+                        Cell::new(child.description),
+                    ]);
+                }
+            } else {
+                eprintln!("warning: no schema found for key '{}'", key);
+            }
         }
 
-        self.validate_toml(&doc).await?;
+        println!("{}", table);
+        Ok(())
+    }
+
+    fn print_diff(&self, old: &str, new: &str) -> EyreResult<()> {
+        let diff = TextDiff::from_lines(old, new);
+        let mut stdout = StandardStream::stdout(ColorChoice::Auto);
 
-        // Save the updated TOML back to the file
-        write(&path, doc.to_string()).await?;
+        for op in diff.ops() {
+            for change in diff.iter_changes(op) {
+                let (sign, color) = match change.tag() {
+                    ChangeTag::Delete => ("-", TermColor::Red),
+                    ChangeTag::Insert => ("+", TermColor::Green),
+                    ChangeTag::Equal => (" ", TermColor::White),
+                };
 
-        info!("Node configuration has been updated");
+                stdout.set_color(ColorSpec::new().set_fg(Some(color)).set_intense(true))?;
+                write!(&mut stdout, "{}{}", sign, change)?;
+            }
```

Comment:
see the `udiff` module

https://docs.rs/similar/latest/similar/udiff/index.html